fulp_modules/features/antagonists/bloodsuckers/code/powers/_powers.dm

```diff
@@ -54,37 +49,34 @@
 	if(power_flags & BP_AM_SINGLEUSE)
 		desc += "<br><br><b>SINGLE USE:</br><i> [name] can only be used once per night.</i>"
 
-/datum/action/bloodsucker/Destroy()
+/datum/action/cooldown/bloodsucker/Destroy()
 	bloodsuckerdatum_power = null
 	return ..()
 
-/datum/action/bloodsucker/IsAvailable(feedback = FALSE)
-	return COOLDOWN_FINISHED(src, bloodsucker_power_cooldown)
-
-/datum/action/bloodsucker/Grant(mob/user)
+/datum/action/cooldown/bloodsucker/Grant(mob/user)
 	. = ..()
 	var/datum/antagonist/bloodsucker/bloodsuckerdatum = IS_BLOODSUCKER(owner)
 	if(bloodsuckerdatum)
 		bloodsuckerdatum_power = bloodsuckerdatum
 
 //This is when we CLICK on the ability Icon, not USING.
-/datum/action/bloodsucker/Trigger(trigger_flags, atom/target)
-	if(active && CheckCanDeactivate()) // Active? DEACTIVATE AND END!
+/datum/action/cooldown/bloodsucker/Trigger(trigger_flags, atom/target)
+	if(active && can_deactivate()) // Active? DEACTIVATE AND END!
 		DeactivatePower()
 		return FALSE
-	if(!CheckCanPayCost() || !CheckCanUse(owner, trigger_flags))
+	if(!can_pay_cost() || !can_use(owner, trigger_flags))
 		return FALSE
-	PayCost()
+	pay_cost()
```

Comment:
the reasoning behind the respelling is mostly what you're going to overwrite to make abilities is capitalized (ActivatePower, Deactivate, etc.) while one time use procs are left uncapitalized